capstone-backend/app.js

```diff
@@ -21,109 +21,170 @@ async function getUserInfo(user){
-    for (var i = 0; i < movies.user_1.length; i++) {
-        user1_genres = user1_genres.concat(movies.user_1[i].genres)
-        user1_movies.push(movies.user_1[i].title)
-    }
-    var user2_genres = []
-    var user2_movies = []
-    for (var i = 0; i < movies.user_2.length; i++) {
-        user2_genres = user2_genres.concat(movies.user_2[i].genres)
-        user2_movies.push(movies.user_2[i].title)
-    }
-    var movieGenreScore = compareArrs(user1_genres, user2_genres, 0.2)
-    var movieTitleScore = compareArrs(user1_movies, user2_movies, 0.05)
-
-    //get tv genre matches
-    var user1_show_genres = []
-    var user1_shows = []
-    for (var i = 0; i < shows.user_1.length; i++) {
-        user1_show_genres = user1_show_genres.concat(shows.user_1[i].genres)
-        user1_shows.push(shows.user_1[i].title)
+function calculateCategoryScore(category, prop1, prop2, weight1, weight2){
+    let user1Prop1 = []
+    let user1Prop2 = []
+    for (let i = 0; i < category.user_1.length; i++) {
+        user1Prop1 = user1Prop1.concat(category.user_1[i].get(prop1))
+        user1Prop2.push(category.user_1[i].get(prop2))
     }
-    var user2_show_genres = []
-    var user2_shows = []
-    for (var i = 0; i < shows.user_2.length; i++) {
-        user2_show_genres = user2_show_genres.concat(shows.user_2[i].genres)
-        user2_shows.push(shows.user_2[i].title)
-    }
-    var showGenreScore = compareArrs(user1_show_genres, user2_show_genres, 0.2)
-    var showTitleScore = compareArrs(user1_shows, user2_shows, 0.05)
-
-    //get hobby matches
-    var user1_hobby_categories = []
-    var user1_hobbies = []
-    for (var i = 0; i < hobbies.user_1.length; i++) {
-        user1_hobby_categories.push(hobbies.user_1[i].category)
-        user1_hobbies.push(hobbies.user_1[i].name)
-    }
-    var user2_hobby_categories = []
-    var user2_hobbies = []
-    for (var i = 0; i < hobbies.user_2.length; i++) {
-        user2_hobby_categories.push(hobbies.user_2[i].category)
-        user2_hobbies.push(hobbies.user_2[i].name)
+
+    let user2Prop1 = []
+    let user2Prop2 = []
+    for (let i = 0; i < category.user_2.length; i++) {
+        user2Prop1 = user2Prop1.concat(category.user_2[i].get(prop1))
+        user2Prop2.push(category.user_2[i].get(prop2))
     }
-    var hobbyCategoryScore = compareArrs(user1_hobby_categories, user2_hobby_categories, 0.25)
-    var hobbyScore = compareArrs(user1_hobbies, user2_hobbies, 0.1)
 
-    var tagsScore = compareArrs(tags.user_1, tags.user_2, 0.15)
-    return (movieGenreScore + movieTitleScore + showGenreScore + showTitleScore + hobbyCategoryScore + hobbyScore + tagsScore)
+    let prop1Score = compareArrs(user1Prop1, user2Prop1, weight1)
+    let prop2Score = compareArrs(user1Prop2, user2Prop2, weight2)
+    return prop1Score + prop2Score
+}
+
+function getScore(movies, shows, hobbies, tags){
+    //get movie genre matches
+    const movieScore = calculateCategoryScore(movies, "genres", "title", 0.2, 0.05)
```

Comment:
It would be good to move the various weights to be a global constant. That way if you ever need to update that value, you can just update the constant. 
example .. add: 

```
const WEIGHT_MOVIE_GENRES = 0.2;
const WEIGHT_MOVIE_TITLE = 0.05;
...
```

to the top of the file